model/project_vars.go

```diff
@@ -556,24 +576,104 @@ func isParameterStoreEnabledForProject(ctx context.Context, projectID string) (b
 		return false, nil
 	}
 
-	projRef, err := FindMergedProjectRef(projectID, "", false)
+	return ref.ParameterStoreEnabled, nil
+}
+
+// findProjectRef finds the project ref associated with the project variables.
+// Returns a bool indicating if it's a branch project ref or a repo ref.
+func (projectVars *ProjectVars) findProjectRef() (ref *ProjectRef, isRepoRef bool, err error) {
+	projectID := projectVars.Id
+	// This intentionally looks for a branch project ref without merging with
+	// its repo ref because project vars for a branch project are stored
+	// separately from project vars for a repo. Therefore, a branch project and
+	// its repo could have differing sync statuses (e.g. it's possible for a
+	// branch project's vars to be synced to Parameter Store, but not its repo
+	// vars).
+	projRef, err := FindBranchProjectRef(projectID)
 	if err != nil {
-		return false, errors.Wrapf(err, "finding merged project ref '%s'", projectID)
+		return nil, false, errors.Wrapf(err, "finding merged project ref '%s'", projectID)
 	}
 	if projRef != nil {
-		return projRef.ParameterStoreEnabled, nil
+		return projRef, false, nil
 	}
 
 	// Project vars could tied to a repo instead of branch project, so check the
 	// repo as a fallback.
 	repoRef, err := FindOneRepoRef(projectID)
 	if err != nil {
-		return false, errors.Wrapf(err, "finding repo ref '%s'", projectID)
+		return nil, false, errors.Wrapf(err, "finding repo ref '%s'", projectID)
 	}
 	if repoRef == nil {
-		return false, errors.Errorf("project or repo ref '%s' not found", projectID)
+		return nil, false, errors.Errorf("project or repo ref '%s' not found", projectID)
+	}
+	return &repoRef.ProjectRef, true, nil
+}
+
+// TODO (DEVPROD-11882): remove full sync logic once the Parameter Store
+// rollout is complete. This functionality only exists to aid the migration
+// process.
+func fullSyncToParameterStore(ctx context.Context, vars *ProjectVars, pRef *ProjectRef, isRepoRef bool) error {
+	grip.Debug(message.Fields{
+		"message":                     "fully syncing project vars to Parameter Store",
+		"num_vars":                    len(vars.Vars),
+		"existing_parameter_mappings": vars.Parameters,
+		"project_id":                  vars.Id,
+		"is_repo_ref":                 isRepoRef,
+		"epic":                        "DEVPROD-5552",
+	})
+
+	// Delete any existing vars to ensure that the project vars are fully synced
+	// starting from a clean state.
+	paramNames := vars.Parameters.ParameterNames()
+	paramMgr := evergreen.GetEnvironment().ParameterManager()
+	if len(paramNames) > 0 {
+		if err := paramMgr.Delete(ctx, paramNames...); err != nil {
+			return errors.Wrap(err, "deleting existing parameters for project vars")
+		}
+	}
+
+	after := vars
+
+	// Since this is fully syncing all the project vars to Parameter Store,
+	// always perform comparisons against a clean state where there are no
+	// parameters.
+	before := &ProjectVars{}
```

Comment:
Can we name this something like "emptyVars" to make it clear that it's not a reflection of the before-deletion state?

Reply:
I think it's already pretty clear because there's a comment right above this saying it's intentionally empty because the initial state is clean. It's also a little clearer to me that `before` is the initial clean state rather than `emptyVars` and is more consistent with how the other upsert function is written.